File: adf_core_python/core/component/gateway/message/am_agent.py
from typing import Any

from rcrs_core.connection import RCRSProto_pb2
from rcrs_core.entities.entity import Entity
from rcrs_core.messages.message import Message
from rcrs_core.worldmodel.entityID import EntityID

from adf_core_python.core.component.gateway.message.urn.urn import (
    ModuleMSG,
    ComponentModuleMSG,
)


class AMAgent(Message):
    def __init__(self) -> None:
        super().__init__(ModuleMSG.AM_AGENT)

    def read(self) -> None:
        pass

    def write(self, agent_id: EntityID, entities: list[Entity]) -> Any:
        entity_proto_list = []
        for entity in entities:
            entity_proto = RCRSProto_pb2.EntityProto()
            entity_proto.urn = entity.get_urn()
            entity_proto.entityID = entity.get_id()

            property_proto_list = []
            for k, v in entity.get_properties().items():
                property_proto_list.append(v.to_property_proto())
            entity_proto.properties.extend(property_proto_list)
            entity_proto_list.append(entity_proto)

        entity_list_proto = RCRSProto_pb2.EntityListProto()
        entity_list_proto.entities.extend(entity_proto_list)
        msg = RCRSProto_pb2.MessageProto()
        msg.urn = self.get_urn()
        msg.components[ComponentModuleMSG.AgentID].entityID = agent_id.get_value()
        msg.components[ComponentModuleMSG.Entities].entityList = entity_list_proto
        return msg

File: java/lib/src/main/java/adf_core_python/gateway/message/AMAgent.java
package adf_core_python.gateway.message;

import adf_core_python.gateway.message.urn.ModuleMessageComponentURN;
import adf_core_python.gateway.message.urn.ModuleMessageURN;
import rescuecore2.config.Config;
import rescuecore2.messages.AbstractMessage;
import rescuecore2.messages.components.ConfigComponent;
import rescuecore2.messages.components.EntityIDComponent;
import rescuecore2.messages.components.EntityListComponent;
import rescuecore2.messages.components.IntComponent;
import rescuecore2.messages.protobuf.RCRSProto;
import rescuecore2.worldmodel.Entity;
import rescuecore2.worldmodel.EntityID;

import java.io.IOException;
import java.io.InputStream;
import java.util.Collection;
import java.util.List;

public class AMAgent extends AbstractMessage {
    private final EntityIDComponent agentID;
    private final EntityListComponent entities;
    private final ConfigComponent config;
    private final IntComponent mode;

    public AMAgent(EntityID agentID, Collection<Entity> entities, Config config, int mode) {
        this();
        this.agentID.setValue(agentID);
        this.entities.setEntities(entities);
        this.config.setConfig(config);
        this.mode.setValue(mode);
    }

    public AMAgent(InputStream inputStream) throws IOException {
        this();
        read(inputStream);
    }

    public AMAgent(RCRSProto.MessageProto messageProto) {
        this();
        fromMessageProto(messageProto);
    }

    private AMAgent() {
        super(ModuleMessageURN.AM_AGENT);
        this.agentID = new EntityIDComponent(ModuleMessageComponentURN.AgentID);
        this.entities = new EntityListComponent(ModuleMessageComponentURN.Entities);
        this.config = new ConfigComponent(ModuleMessageComponentURN.Config);
        this.mode = new IntComponent(ModuleMessageComponentURN.Mode);
        addMessageComponent(agentID);
        addMessageComponent(entities);
        addMessageComponent(config);
        addMessageComponent(mode);
    }

    public EntityID getAgentID() {
        return this.agentID.getValue();
    }

    public List<Entity> getEntities() {
        return this.entities.getEntities();
    }

    public Config getConfig() {
        return this.config.getConfig();
    }

    public int getMode() {
        return this.mode.getValue();
    }
}
